pkg/location/type_dir.go

```diff
@@ -0,0 +1,33 @@
+package location
+
+import (
+	"fmt"
+	"io/fs"
+)
+
+type Dir struct {
+	Directory string
+}
+
+var _ Reference = Dir{}
+
+func parseDir(location string, opt options) (Reference, error) {
+	if fs.ValidPath(location) {
```

Comment:
does this operate in the context of local filesystem, or is this purely stateless validation analyzing only the input string?